pkg/utils/docker/push.go

```diff
@@ -59,7 +59,16 @@ func (c *Push) PushImage(fullImageName string) error {
 	// $DOCKER_CONFIG/config.json, $HOME/.docker/config.json , $HOME/.dockercfg
 	credentials, err := dockerlib.NewAuthConfigurationsFromDockerCfg()
 	if err != nil {
-		return errors.Wrap(err, "Unable to retrieve .docker/config.json authentication details. Check that 'docker login' works successfully on the command line.")
+		log.Warn(errors.Wrap(err, "Unable to retrieve .docker/config.json authentication details. Check that 'docker login' works successfully on the command line."))
+	}
+
+	if credentials == nil {
```

Comment:
Mind adding a comment here saying TODO: refactor as well as detail the error (add the fsouza/dockerclient-go) issue? Otherwise, this code LGTM.

Reply:
The thing is that it would be nice to have this approach (try push without authentication) cause when you use local private repo it doesn't actually require login (via docker login command) there before accessing it. 

If you thing kompose should always expect available credentials in config.json I can add the comment "TODO: refactor as well as dockerclient-go starts retrieve credentials from osxkeychain (add the fsouza/dockerclient-go) issue".

What do you think?